### ISSUE TITLE
Error during update

### ISSUE THREAD

**Issue by pikeas** [sorin-ionescu/prezto#845]
Following the recommended update procedure:

    ~/.zprezto git pull && git submodule update --init --recursive
    Already up-to-date.
    fatal: reference is not a tree: d0381d859c0eb8d52a6211281e622e89153ef812
    Unable to checkout 'd0381d859c0eb8d52a6211281e622e89153ef812' in submodule path 'modules/history-substring-search/external'

**Comment by nasenatmer** [comment]
Same here. Updating prezto (`gfm upstream master`) and submodules (`gSI`) throws the same error here.

**Comment by sorin-ionescu** [comment]
When I clone from scratch, I don't have a problem initializing modules. At what commit were you before trying to update; so, that I may attempt to replicate it?

**Comment by pzuraq** [comment]
Getting the same issue when trying to clone from scratch:

```
Cloning into '/users/pzuraq/.zprezto'...
remote: Counting objects: 8495, done.
remote: Total 8495 (delta 0), reused 0 (delta 0), pack-reused 8495
Receiving objects: 100% (8495/8495), 1.69 MiB | 445.00 KiB/s, done.
Resolving deltas: 100% (3760/3760), done.
Checking connectivity... done.
Submodule 'modules/completion/external' (https://github.com/zsh-users/zsh-completions.git) registered for path 'modules/completion/external'
Submodule 'modules/history-substring-search/external' (https://github.com/zsh-users/zsh-history-substring-search.git) registered for path 'modules/history-substring-search/external'
Submodule 'modules/prompt/external/agnoster' (https://gist.github.com/3712874.git) registered for path 'modules/prompt/external/agnoster'
Submodule 'modules/prompt/external/powerline' (https://github.com/davidjrice/prezto_powerline.git) registered for path 'modules/prompt/external/powerline'
Submodule 'modules/prompt/functions/pure' (https://github.com/sindresorhus/pure.git) registered for path 'modules/prompt/external/pure'
Submodule 'modules/syntax-highlighting/external' (https://github.com/zsh-users/zsh-syntax-highlighting.git) registered for path 'modules/syntax-highlighting/external'
Cloning into 'modules/completion/external'...
remote: Counting objects: 2235, done.
remote: Total 2235 (delta 0), reused 0 (delta 0), pack-reused 2235
Receiving objects: 100% (2235/2235), 888.25 KiB | 335.00 KiB/s, done.
Resolving deltas: 100% (1288/1288), done.
Checking connectivity... done.
Submodule path 'modules/completion/external': checked out '68c949470eaa87c5d67080d32fb2b69c8b837eb4'
Cloning into 'modules/history-substring-search/external'...
remote: Counting objects: 261, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 261 (delta 0), reused 0 (delta 0), pack-reused 258
Receiving objects: 100% (261/261), 78.24 KiB | 112.00 KiB/s, done.
Resolving deltas: 100% (128/128), done.
Checking connectivity... done.
fatal: reference is not a tree: d0381d859c0eb8d52a6211281e622e89153ef812
Cloning into 'modules/prompt/external/agnoster'...
remote: Counting objects: 71, done.
remote: Compressing objects: 100% (62/62), done.
remote: Total 71 (delta 23), reused 38 (delta 8), pack-reused 0
Unpacking objects: 100% (71/71), done.
Checking connectivity... done.
Submodule path 'modules/prompt/external/agnoster': checked out '43cb371f361eecf62e9dac7afc73a1c16edf89c7'
Cloning into 'modules/prompt/external/powerline'...
remote: Counting objects: 22, done.
remote: Total 22 (delta 0), reused 0 (delta 0), pack-reused 22
Unpacking objects: 100% (22/22), done.
Checking connectivity... done.
Submodule path 'modules/prompt/external/powerline': checked out '8e81152340c4beb2d941340d1feb2dc29bbcc309'
Cloning into 'modules/prompt/external/pure'...
remote: Counting objects: 347, done.
remote: Total 347 (delta 0), reused 0 (delta 0), pack-reused 347
Receiving objects: 100% (347/347), 250.28 KiB | 54.00 KiB/s, done.
Resolving deltas: 100% (153/153), done.
Checking connectivity... done.
Submodule path 'modules/prompt/external/pure': checked out 'ef1329909a3c72bb76765ccc519214ce60cedddb'
Cloning into 'modules/syntax-highlighting/external'...
remote: Counting objects: 828, done.
remote: Total 828 (delta 0), reused 0 (delta 0), pack-reused 828
Receiving objects: 100% (828/828), 140.66 KiB | 173.00 KiB/s, done.
Resolving deltas: 100% (440/440), done.
Checking connectivity... done.
Submodule path 'modules/syntax-highlighting/external': checked out '3dc574190071ef59054f6d693132410b37aa3ae4'
Unable to checkout 'd0381d859c0eb8d52a6211281e622e89153ef812' in submodule path 'modules/history-substring-search/external'
```

**Comment by sorin-ionescu** [comment]
I do not know. It works here with the latest `git`. Try `cd` into that module then `checkout master`. Then update it.

**Comment by pzuraq** [comment]
My computer is running Mavericks, could that be an issue?

**Comment by pikeas** [comment]
~/.z/m/h/external git:master~1 gco master
    Previous HEAD position was 9f9fc7d... README: MacOSX 10.9 users need to bind ^[[A/B too
    Switched to branch 'master'
    Your branch is behind 'origin/master' by 7 commits, and can be fast-forwarded.
      (use "git pull" to update your local branch)
    ~/.z/m/h/external git:master git pull
    Updating 3c7b079..6d7bc71
    Fast-forward
     README.md                        |  8 ++++++++
     zsh-history-substring-search.zsh | 34 +++++++++++++++++++++++++++++-----
     2 files changed, 37 insertions(+), 5 deletions(-)
    ~/.z/m/h/external git:master cd ../../..
    ~/.zprezto git:master git pull && git submodule update --init --recursive
    Already up-to-date.
    fatal: reference is not a tree: d0381d859c0eb8d52a6211281e622e89153ef812
    Unable to checkout 'd0381d859c0eb8d52a6211281e622e89153ef812' in submodule path 'history-substring-search/external'

**Comment by Eriner** [comment]
This is not a platform-specific issue.

**Comment by sorin-ionescu** [comment]
Fixed in c98da90.
